// @spectrum-web-components/core/dist/shared/base/Base.js
import { LitElement as E } from "lit";
import { version as v } from "./version.js";
var w, g, h;
const u = /* @__PURE__ */ new Set(), b = () => {
  const i = document.documentElement.dir === "rtl" ? document.documentElement.dir : "ltr";
  u.forEach((d) => {
    d.setAttribute("dir", i);
  });
}, N = new MutationObserver(b);
N.observe(document.documentElement, {
  attributes: !0,
  attributeFilter: ["dir"]
});
const W = (i) => typeof i.startManagingContentDirection != "undefined" || i.tagName === "SP-THEME";
function C(i) {
  class d extends i {
    /**
     * @internal
     */
    get isLTR() {
      return this.dir === "ltr";
    }
    /**
     * @internal
     */
    hasVisibleFocusInTree() {
      const n = ((r = document) => {
        var s;
        let e = r.activeElement;
        for (; e != null && e.shadowRoot && e.shadowRoot.activeElement; )
          e = e.shadowRoot.activeElement;
        const a = e ? [e] : [];
        for (; e; ) {
          const c = e.assignedSlot || e.parentElement || ((s = e.getRootNode()) == null ? void 0 : s.host);
          c && a.push(c), e = c;
        }
        return a;
      })(
        this.getRootNode()
      )[0];
      if (!n)
        return !1;
      try {
        return n.matches(":focus-visible") || n.matches(".focus-visible");
      } catch (r) {
        return n.matches(".focus-visible");
      }
    }
    connectedCallback() {
      if (!this.hasAttribute("dir")) {
        let t = this.assignedSlot || this.parentNode;
        for (; t !== document.documentElement && !W(
          t
        ); )
          t = t.assignedSlot || // step into the shadow DOM of the parent of a slotted node
          t.parentNode || // DOM Element detected
          t.host;
        if (this.dir = t.dir === "rtl" ? t.dir : this.dir || "ltr", t === document.documentElement)
          u.add(this);
        else {
          const { localName: n } = t;
          n.search("-") > -1 && !customElements.get(n) ? customElements.whenDefined(n).then(() => {
            t.startManagingContentDirection(this);
          }) : t.startManagingContentDirection(
            this
          );
        }
        this._dirParent = t;
      }
      super.connectedCallback();
    }
    disconnectedCallback() {
      super.disconnectedCallback(), this._dirParent && (this._dirParent === document.documentElement ? u.delete(this) : this._dirParent.stopManagingContentDirection(
        this
      ), this.removeAttribute("dir"));
    }
  }
  return d;
}
class L extends C(E) {
}
L.VERSION = v;
if (process.env.NODE_ENV === "development") {
  const i = {
    default: !1,
    accessibility: !1,
    api: !1
  }, d = {
    default: !1,
    low: !1,
    medium: !1,
    high: !1,
    deprecation: !1
  };
  window.__swc = {
    ...window.__swc,
    DEBUG: !0,
    ignoreWarningLocalNames: {
      ...((w = window.__swc) == null ? void 0 : w.ignoreWarningLocalNames) || {}
    },
    ignoreWarningTypes: {
      ...i,
      ...((g = window.__swc) == null ? void 0 : g.ignoreWarningTypes) || {}
    },
    ignoreWarningLevels: {
      ...d,
      ...((h = window.__swc) == null ? void 0 : h.ignoreWarningLevels) || {}
    },
    issuedWarnings: /* @__PURE__ */ new Set(),
    warn: (o, t, n, { type: r = "api", level: s = "default", issues: e } = {}) => {
      const { localName: a = "base" } = o || {}, c = `${a}:${r}:${s}`;
      if (!window.__swc.verbose && window.__swc.issuedWarnings.has(c) || window.__swc.ignoreWarningLocalNames[a] || window.__swc.ignoreWarningTypes[r] || window.__swc.ignoreWarningLevels[s])
        return;
      window.__swc.issuedWarnings.add(c);
      let m = "";
      e && e.length && (e.unshift(""), m = e.join(`
    - `) + `
`);
      const f = s === "deprecation" ? "DEPRECATION NOTICE: " : "", p = o ? `
Inspect this issue in the follow element:` : "", _ = (o ? `

` : `
`) + n + `
`, l = [];
      l.push(
        f + t + `
` + m + p
      ), o && l.push(o), l.push(_, {
        data: {
          localName: a,
          type: r,
          level: s
        }
      }), console.warn(...l);
    }
  }, window.__swc.warn(
    void 0,
    "Spectrum Web Components is in dev mode. Not recommended for production!",
    "https://opensource.adobe.com/spectrum-web-components/dev-mode/",
    { type: "default" }
  );
}
export {
  L as SpectrumElement,
  C as SpectrumMixin
};
