// @spectrum-web-components/core/dist/shared/base/sizedMixin.js
import { property as c } from "lit/decorators.js";
var u = Object.defineProperty, h = Object.getOwnPropertyDescriptor, _ = (r, t, i, n) => {
  for (var s = h(t, i), o = r.length - 1, e; o >= 0; o--)
    (e = r[o]) && (s = e(t, i, s) || s);
  return s && u(t, i, s), s;
};
const a = {
  xxs: "xxs",
  xs: "xs",
  s: "s",
  m: "m",
  l: "l",
  xl: "xl",
  xxl: "xxl"
};
function d(r, {
  validSizes: t = ["s", "m", "l", "xl"],
  noDefaultSize: i,
  defaultSize: n = "m"
} = {}) {
  class s extends r {
    constructor() {
      super(...arguments), this._size = n;
    }
    get size() {
      return this._size || n;
    }
    set size(e) {
      const x = i ? null : n, l = e && e.toLocaleLowerCase(), p = t.includes(l) ? l : x;
      if (p && this.setAttribute("size", p), this._size === p)
        return;
      const z = this._size;
      this._size = p, this.requestUpdate("size", z);
    }
    update(e) {
      !this.hasAttribute("size") && !i && this.setAttribute("size", this.size), super.update(e);
    }
  }
  return s.VALID_SIZES = t, _([
    c({ type: String })
  ], s.prototype, "size"), s;
}
export {
  a as ElementSizes,
  d as SizedMixin
};
