// @spectrum-web-components/core/dist/components/progress-circle/ProgressCircle.base.js
import { property as s, query as b } from "lit/decorators.js";
import { SizedMixin as d, SpectrumElement as u } from "@spectrum-web-components/core/shared/base/index.js";
import { getLabelFromSlot as h } from "@spectrum-web-components/core/shared/get-label-from-slot.js";
import { PROGRESS_CIRCLE_VALID_SIZES as c } from "./ProgressCircle.types.js";
var m = Object.defineProperty, i = (a, e, r, o) => {
  for (var t = void 0, p = a.length - 1, n; p >= 0; p--)
    (n = a[p]) && (t = n(e, r, t) || t);
  return t && m(e, r, t), t;
};
class l extends d(u, {
  validSizes: c
}) {
  constructor() {
    super(...arguments), this.indeterminate = !1, this.label = "", this.progress = 0;
  }
  makeRotation(e) {
    return this.indeterminate ? void 0 : `transform: rotate(${e}deg);`;
  }
  handleSlotchange() {
    const e = h(this.label, this.slotEl);
    e && (this.label = e);
  }
  firstUpdated(e) {
    super.firstUpdated(e), this.hasAttribute("role") || this.setAttribute("role", "progressbar");
  }
  updated(e) {
    var r, o;
    super.updated(e), !this.indeterminate && e.has("progress") ? this.setAttribute("aria-valuenow", "" + this.progress) : this.hasAttribute("aria-valuenow") && this.removeAttribute("aria-valuenow"), e.has("label") && (this.label.length ? this.setAttribute("aria-label", this.label) : e.get("label") === this.getAttribute("aria-label") && this.removeAttribute("aria-label"));
    const t = () => !!(this.label || this.getAttribute("aria-label") || this.getAttribute("aria-labelledby") || this.slotEl.assignedNodes().length);
    (r = window.__swc) != null && r.DEBUG && !t() && this.getAttribute("role") === "progressbar" && ((o = window.__swc) == null || o.warn(
      this,
      "<sp-progress-circle> elements need one of the following to be accessible:",
      "https://opensource.adobe.com/spectrum-web-components/components/progress-circle/#accessibility",
      {
        type: "accessibility",
        issues: [
          'value supplied to the "label" attribute, which will be displayed visually as part of the element, or',
          "text content supplied directly to the <sp-progress-circle> element, or",
          'value supplied to the "aria-label" attribute, which will only be provided to screen readers, or',
          'an element ID reference supplied to the "aria-labelledby" attribute, which will be provided by screen readers and will need to be managed manually by the parent application.'
        ]
      }
    ));
  }
}
i([
  s({ type: String, reflect: !0, attribute: "static-color" })
], l.prototype, "staticColor");
i([
  s({ type: Boolean, reflect: !0 })
], l.prototype, "indeterminate");
i([
  s({ type: String })
], l.prototype, "label");
i([
  s({ type: Number })
], l.prototype, "progress");
i([
  b("slot")
], l.prototype, "slotEl");
export {
  l as ProgressCircleBase
};
