// @spectrum-web-components/core/dist/components/alert-banner/AlertBanner.base.js
import { property as o } from "lit/decorators.js";
import { SpectrumElement as h } from "@spectrum-web-components/core/shared/base/index.js";
var c = Object.defineProperty, d = Object.getOwnPropertyDescriptor, l = (r, e, s, i) => {
  for (var t = i > 1 ? void 0 : i ? d(e, s) : e, n = r.length - 1, a; n >= 0; n--)
    (a = r[n]) && (t = (i ? a(e, s, t) : a(t)) || t);
  return i && t && c(e, s, t), t;
};
const u = ["neutral", "info", "negative"];
class p extends h {
  constructor() {
    super(...arguments), this.open = !1, this.dismissible = !1, this._variant = "";
  }
  static get styles() {
    return [];
  }
  set variant(e) {
    var s;
    if (e === this.variant)
      return;
    const i = this.variant;
    this.isValidVariant(e) ? (this.setAttribute("variant", e), this._variant = e) : (this.removeAttribute("variant"), this._variant = "", (s = window.__swc) != null && s.DEBUG && window.__swc.warn(
      this,
      `<${this.localName}> element expects the "variant" attribute to be one of the following:`,
      "https://opensource.adobe.com/spectrum-web-components/components/alert-banner/#variants",
      {
        issues: [...u]
      }
    )), this.requestUpdate("variant", i);
  }
  get variant() {
    return this._variant;
  }
  isValidVariant(e) {
    return u.includes(e);
  }
  shouldClose() {
    this.dispatchEvent(
      new CustomEvent("close", {
        composed: !0,
        bubbles: !0,
        cancelable: !0
      })
    ) && this.close();
  }
  close() {
    this.open = !1;
  }
  handleKeydown(e) {
    e.code === "Escape" && this.dismissible && this.shouldClose();
  }
  updated(e) {
    super.updated(e), e.has("open") && (this.open ? this.addEventListener("keydown", this.handleKeydown) : this.removeEventListener("keydown", this.handleKeydown));
  }
}
l([
  o({ type: Boolean, reflect: !0 })
], p.prototype, "open", 2);
l([
  o({ type: Boolean, reflect: !0 })
], p.prototype, "dismissible", 2);
l([
  o({ type: String })
], p.prototype, "variant", 1);
export {
  p as AlertBannerBase
};
