// @spectrum-web-components/core/dist/components/status-light/StatusLight.base.js
import { property as a } from "lit/decorators.js";
import { SizedMixin as p, SpectrumElement as u } from "@spectrum-web-components/core/shared/base";
var c = Object.defineProperty, d = (o, s, e, i) => {
  for (var t = void 0, r = o.length - 1, n; r >= 0; r--)
    (n = o[r]) && (t = n(s, e, t) || t);
  return t && c(s, e, t), t;
};
class l extends p(u, {
  noDefaultSize: !0
}) {
  constructor() {
    super(...arguments), this.variant = "info";
  }
  // ──────────────────────
  //     IMPLEMENTATION
  // ──────────────────────
  updated(s) {
    var e;
    if (super.updated(s), (e = window.__swc) != null && e.DEBUG) {
      const i = this.constructor;
      i.VARIANTS.includes(this.variant) || window.__swc.warn(
        this,
        `<${this.localName}> element expects the "variant" attribute to be one of the following:`,
        "https://opensource.adobe.com/spectrum-web-components/components/status-light/#variants",
        {
          issues: [...i.VARIANTS]
        }
      ), this.hasAttribute("disabled") && !("disabled" in this) && window.__swc.warn(
        this,
        `<${this.localName}> element does not support the disabled state.`,
        "https://opensource.adobe.com/spectrum-web-components/components/status-light/#states",
        {
          issues: [
            "disabled is not a supported property in Spectrum 2"
          ]
        }
      );
    }
  }
}
d([
  a({ type: String, reflect: !0 })
], l.prototype, "variant");
export {
  l as StatusLightBase
};
