// @spectrum-web-components/core/dist/components/divider/Divider.base.js
import { property as p } from "lit/decorators.js";
import { SizedMixin as u, SpectrumElement as f } from "@spectrum-web-components/core/shared/base/index.js";
import { DIVIDER_VALID_SIZES as d, DIVIDER_STATIC_COLORS as n } from "./Divider.types.js";
var c = Object.defineProperty, l = (r, t, o, v) => {
  for (var e = void 0, i = r.length - 1, s; i >= 0; i--)
    (s = r[i]) && (e = s(t, o, e) || e);
  return e && c(t, o, e), e;
};
class a extends u(f, {
  validSizes: d,
  /**@todo the design spec says the default size is small but we declare no default size */
  noDefaultSize: !0
}) {
  constructor() {
    super(...arguments), this.vertical = !1;
  }
  firstUpdated(t) {
    super.firstUpdated(t), this.setAttribute("role", "separator");
  }
  updated(t) {
    super.updated(t), t.has("vertical") && (this.vertical ? this.setAttribute("aria-orientation", "vertical") : this.removeAttribute("aria-orientation"));
  }
}
a.STATIC_COLORS = n;
l([
  p({ type: Boolean, reflect: !0 })
], a.prototype, "vertical");
l([
  p({ reflect: !0, attribute: "static-color" })
], a.prototype, "staticColor");
export {
  a as DividerBase
};
