// @spectrum-web-components/core/dist/shared/observe-slot-presence.js
import { MutationController as g } from "@lit-labs/observers/mutation-controller.js";
const e = Symbol("slotContentIsPresent");
function b(a, r) {
  var i, l;
  const n = Array.isArray(r) ? r : [r];
  class c extends (l = a, i = e, l) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    constructor(...t) {
      super(...t), this[i] = /* @__PURE__ */ new Map(), this.managePresenceObservedSlot = () => {
        let o = !1;
        n.forEach((s) => {
          const h = !!this.querySelector(`:scope > ${s}`), u = this[e].get(s) || !1;
          o = o || u !== h, this[e].set(
            s,
            !!this.querySelector(`:scope > ${s}`)
          );
        }), o && this.updateComplete.then(() => {
          this.requestUpdate();
        });
      }, new g(this, {
        config: {
          childList: !0,
          subtree: !0
        },
        callback: () => {
          this.managePresenceObservedSlot();
        }
      }), this.managePresenceObservedSlot();
    }
    /**
     *  @internal
     */
    get slotContentIsPresent() {
      if (n.length === 1)
        return this[e].get(n[0]) || !1;
      throw new Error(
        "Multiple selectors provided to `ObserveSlotPresence` use `getSlotContentPresence(selector: string)` instead."
      );
    }
    /**
     * @internal
     */
    getSlotContentPresence(t) {
      if (this[e].has(t))
        return this[e].get(t) || !1;
      throw new Error(
        `The provided selector \`${t}\` is not being observed.`
      );
    }
  }
  return c;
}
export {
  b as ObserveSlotPresence
};
