// @spectrum-web-components/core/dist/components/badge/Badge.base.js
import { property as c } from "lit/decorators.js";
import { SizedMixin as u, SpectrumElement as l } from "@spectrum-web-components/core/shared/base/index.js";
import { ObserveSlotPresence as f } from "@spectrum-web-components/core/shared/observe-slot-presence.js";
import { ObserveSlotText as m } from "@spectrum-web-components/core/shared/observe-slot-text.js";
import { FIXED_VALUES as h, BADGE_VARIANTS_SEMANTIC as d } from "./Badge.types.js";
var _ = Object.defineProperty, v = Object.getOwnPropertyDescriptor, p = (o, t, r, e) => {
  for (var s = e > 1 ? void 0 : e ? v(t, r) : t, n = o.length - 1, a; n >= 0; n--)
    (a = o[n]) && (s = (e ? a(t, r, s) : a(s)) || s);
  return e && s && _(t, r, s), s;
};
class i extends u(
  m(f(l, '[slot="icon"]'), ""),
  {
    noDefaultSize: !0
  }
) {
  constructor() {
    super(...arguments), this.variant = "informative";
  }
  get fixed() {
    return this._fixed;
  }
  set fixed(t) {
    if (t === this.fixed)
      return;
    const r = this.fixed;
    this._fixed = t, t ? this.setAttribute("fixed", t) : this.removeAttribute("fixed"), this.requestUpdate("fixed", r);
  }
  // ──────────────────────
  //     IMPLEMENTATION
  // ──────────────────────
  /**
   * @internal Used for rendering gap when the badge has an icon.
   */
  get hasIcon() {
    return this.slotContentIsPresent;
  }
  update(t) {
    var r;
    if (super.update(t), (r = window.__swc) != null && r.DEBUG) {
      const e = this.constructor;
      e.VARIANTS.includes(this.variant) || window.__swc.warn(
        this,
        `<${this.localName}> element expect the "variant" attribute to be one of the following:`,
        "https://opensource.adobe.com/spectrum-web-components/components/badge/#variants",
        {
          issues: [...e.VARIANTS]
        }
      ), "outline" in this && this.outline === !0 && !e.VARIANTS_SEMANTIC.includes(this.variant) && window.__swc.warn(
        this,
        `<${this.localName}> element only supports the outline styling if the variant is a semantic color variant.`,
        "https://opensource.adobe.com/spectrum-web-components/components/badge/#variants",
        {
          issues: [...e.VARIANTS_SEMANTIC]
        }
      );
    }
  }
}
i.FIXED_VALUES = h;
i.VARIANTS_SEMANTIC = d;
p([
  c({ type: String, reflect: !0 })
], i.prototype, "variant", 2);
p([
  c({ reflect: !0 })
], i.prototype, "fixed", 1);
export {
  i as BadgeBase
};
